### PR TITLE
Continue removal of custom script code in ClassNode in Painless

### PR DESCRIPTION
**Pull request by jdconrad**
This continues down the path of removing custom code required for Elasticsearch Painless scripts from the user and ir nodes. This change uses ir nodes to inject gets methods, needs methods, and the bootstrap method as required. There is no reason for the SClass/ClassNode to know about these specific items.

Though, these appear as separate phases for now, this is purely transitional. Eventually, these will be rolled into a single "build" phase where the user tree is semantically checked and converted into an ir tree. This intermediate step is necessary to make the changes occur in smaller consumable steps.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/ClassNode.java [hunk 1]
```diff
@@ -37,22 +37,10 @@
...
 import java.util.Collection;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 import static org.elasticsearch.painless.WriterConstants.BASE_INTERFACE_TYPE;
-import static org.elasticsearch.painless.WriterConstants.BITSET_TYPE;
 import static org.elasticsearch.painless.WriterConstants.CLASS_TYPE;
-import static org.elasticsearch.painless.WriterConstants.DEFINITION_TYPE;
-import static org.elasticsearch.painless.WriterConstants.DEF_BOOTSTRAP_DELEGATE_METHOD;
```
Comment (rjernst) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/ClassNode.java]: We can remove these constants from the code now right?

**Review by rjernst — APPROVED** [review]
LGTM

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/ClassNode.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/ClassNode.java]: Yes! Good catch.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review!

**Review by stu-elastic — APPROVED** [review]
lgtm thanks for walking me through this.

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review.

**Comment by jdconrad** [comment]
Relates to https://github.com/elastic/elasticsearch/issues/49869

**Pull request merged**